nixos/modules/services/networking/sslh.nix

```diff
@@ -56,9 +60,9 @@ in
       };
 
       listenAddress = mkOption {
-        type = types.str;
-        default = "0.0.0.0";
-        description = "Listening address or hostname.";
+        type = types.coercedTo types.str singleton (types.listOf types.str);
+        default = [ "0.0.0.0" "[::]" ];
```

Comment:
I thought if you listened on `[::]` it would automatically include the IPv4-subset of IPv6, and listen on IPv6 and IPv4?

Reply:
I tried with the test and it does not work.
```diff
diff --git a/nixos/tests/sslh.nix b/nixos/tests/sslh.nix
index 8f8b4a83578..7cc3ece152a 100644
--- a/nixos/tests/sslh.nix
+++ b/nixos/tests/sslh.nix
@@ -18,6 +18,7 @@ import ./make-test-python.nix {
       services.sslh = {
         enable = true;
         transparent = true;
+        listenAddress = "[::]";
         appendConfig = ''
           protocols:
           (
```
yields: 
```
client: must succeed: ssh -4 -p 443 -i /etc/sshKey -o StrictHostKeyChecking=accept-new server 'touch /tmp/foo'
client # ssh: connect to host server port 443: Connection refused
client: output:
```

Reply:
Seems surprising to people on IRC, too:


```
14:06 <bennofs> gchristensen: maybe try just 0? i have no idea if that works though
14:06 <hexa-> gchristensen: depends on net.ipv6.bindv6only
14:07 <hexa-> should be 0 though
```

I wonder if `sslh` has a bug preventing this from working.

Reply:
Found it: https://github.com/yrutschle/sslh/blob/49e8bff01bb544aea881622c593ca022ee820721/common.c#L173

IMO this should probably not be the case. Maybe the maintainers here know something.

Reply:
https://github.com/yrutschle/sslh/pull/161

Reply:
In any case: it is legitimate to want to listen on several ip addresses. Wanting ipv6 is only one of the possible use cases.
Given that the sslh config file does not allow overriding, the module should expose this possibility.
Also, the increase in complexity is quite low, so I think we should not focus on making things work with the old module.